### PR TITLE
feat(flat-components): add invite code support

### PR DESCRIPTION
**Pull request by Cheerego7**
- add inviteCode in flatServers

### CODE REVIEW

**Review by hyrious — DISMISSED** [review]

File: desktop/renderer-app/src/apiMiddleware/flatServer/index.ts [hunk 1]
```diff
@@ -89,6 +89,8 @@ export interface FlatServerRoom {
     periodicUUID: string | null;
     /** 房间所有者的 uuid */
     ownerUUID: string;
+    /** 房间邀请码 */
```
Comment (BlackHole1) [inline desktop/renderer-app/src/apiMiddleware/flatServer/index.ts]: to english

File: desktop/renderer-app/src/pages/HomePage/MainRoomMenu/JoinRoomBox.tsx [hunk 2]
```diff
@@ -178,7 +178,7 @@ export const JoinRoomBox = observer<JoinRoomBoxProps>(function JoinRoomBox({ onJ
             const values = form.getFieldsValue();
             configStore.updateAutoMicOn(values.autoMicOn);
             configStore.updateAutoCameraOn(values.autoCameraOn);
-            await sp(onJoinRoom(values.roomUUID));
+            await sp(onJoinRoom(values.roomUUID.replace(/\s+/g, "")));
```
Comment (BlackHole1) [inline desktop/renderer-app/src/pages/HomePage/MainRoomMenu/JoinRoomBox.tsx]: put the `replace` operation in `onJoinRoom`

File: desktop/renderer-app/src/stores/RoomStore.ts [hunk 3]
```diff
@@ -31,6 +31,8 @@ export interface RoomItem {
     roomUUID: string;
     /** 房间所有者的 uuid */
     ownerUUID: string;
+    /** 房间邀请码 */
```
Comment (BlackHole1) [inline desktop/renderer-app/src/stores/RoomStore.ts]: to english

File: packages/flat-components/src/components/InviteModal/index.tsx [hunk 4]
```diff
@@ -31,9 +31,17 @@ export const InviteModal: React.FC<InviteModalProps> = ({
...
     const uuid = periodicUUID || roomUUID;
 
+    let roomInviteCode: string;
+    if (inviteCode && /^\d{10}$/.test(inviteCode)) {
+        roomInviteCode =
+            inviteCode.slice(0, 3) + " " + inviteCode.slice(3, 6) + " " + inviteCode.slice(6);
+    } else {
+        roomInviteCode = uuid;
+    }
+
```
Comment (BlackHole1) [inline packages/flat-components/src/components/InviteModal/index.tsx]: use css content

File: packages/flat-components/src/components/RoomDetailPage/RoomDetailBody/index.tsx [hunk 5]
```diff
@@ -23,9 +23,17 @@ export const RoomDetailBody = observer<RoomDetailBodyProps>(function RoomDetailB
...
     const lang = i18n.language;
 
+    let roomInviteCode: string;
+    if (inviteCode && /^\d{10}$/.test(inviteCode)) {
+        roomInviteCode =
+            inviteCode.slice(0, 3) + " " + inviteCode.slice(3, 6) + " " + inviteCode.slice(6);
+    } else {
+        roomInviteCode = periodicUUID || roomUUID;
+    }
+
```
Comment (BlackHole1) [inline packages/flat-components/src/components/RoomDetailPage/RoomDetailBody/index.tsx]: use content

File: packages/flat-components/src/types/room.ts [hunk 6]
```diff
@@ -3,6 +3,8 @@ export type RoomInfo = {
     roomUUID: string;
     /** 房间所有者的 uuid */
     ownerUUID: string;
+    /** 房间邀请码 */
```
Comment (BlackHole1) [inline packages/flat-components/src/types/room.ts:6]: to english

File: web/flat-web/src/apiMiddleware/flatServer/index.ts [hunk 7]
```diff
@@ -89,6 +90,8 @@ export interface FlatServerRoom {
     periodicUUID: string | null;
     /** 房间所有者的 uuid */
     ownerUUID: string;
+    /** 房间邀请码 */
```
Comment (BlackHole1) [inline web/flat-web/src/apiMiddleware/flatServer/index.ts]: to english

File: web/flat-web/src/pages/HomePage/MainRoomMenu/JoinRoomBox.tsx [hunk 8]
```diff
@@ -178,7 +178,7 @@ export const JoinRoomBox = observer<JoinRoomBoxProps>(function JoinRoomBox({ onJ
             const values = form.getFieldsValue();
             configStore.updateAutoMicOn(values.autoMicOn);
             configStore.updateAutoCameraOn(values.autoCameraOn);
-            await sp(onJoinRoom(values.roomUUID));
+            await sp(onJoinRoom(values.roomUUID.replaceAll(" ", "")));
```
Comment (BlackHole1) [inline web/flat-web/src/pages/HomePage/MainRoomMenu/JoinRoomBox.tsx]: put the `replace` operation in `onJoinRoom` to complete

File: web/flat-web/src/stores/RoomStore.ts [hunk 9]
```diff
@@ -31,6 +31,8 @@ export interface RoomItem {
     roomUUID: string;
     /** 房间所有者的 uuid */
     ownerUUID: string;
+    /** 房间邀请码 */
```
Comment (BlackHole1) [inline web/flat-web/src/stores/RoomStore.ts]: to english

File: packages/flat-components/src/components/InviteModal/index.tsx [hunk 10]
```diff
@@ -31,9 +31,17 @@ export const InviteModal: React.FC<InviteModalProps> = ({
     onCancel,
 }) => {
     const { t, i18n } = useTranslation();
-    const { beginTime, endTime, periodicUUID, roomUUID, title } = room;
+    const { beginTime, endTime, periodicUUID, roomUUID, inviteCode, title } = room;
     const uuid = periodicUUID || roomUUID;
 
+    let roomInviteCode: string;
+    if (inviteCode && /^\d{10}$/.test(inviteCode)) {
```
Comment (BlackHole1) [inline packages/flat-components/src/components/InviteModal/index.tsx]: inviteCode definitely exists

**Review by BlackHole1 — CHANGES_REQUESTED** [review]

File: packages/flat-components/src/utils/room.ts [hunk 11]
```diff
@@ -189,3 +189,15 @@ export function roomStatusToI18nKey(
...
+ * format invite code
+ * @example
+ * 123456789 -> 123 456 7890
+ */
+export const formatInviteCode = (inviteCode: string): string => {
+    if (/^\d{10}$/.test(inviteCode)) {
+        return inviteCode.slice(0, 3) + " " + inviteCode.slice(3, 6) + " " + inviteCode.slice(6);
+    }
+    return inviteCode;
+};
```
Comment (BlackHole1) [inline packages/flat-components/src/utils/room.ts]:
```suggestion
export const formatInviteCode = (inviteCode: string, uuid: string): string => {
    if (/^\d{10}$/.test(inviteCode)) {
        // 123456789 -> 123 456 7890
        return inviteCode.slice(0, 3) + " " + inviteCode.slice(3, 6) + " " + inviteCode.slice(6);
    }
    return uuid;
};
```

**Review by BlackHole1 — CHANGES_REQUESTED** [review]

**Review by BlackHole1 — APPROVED** [review]

**Pull request merged**